Website/Providers/DataProviders/SqlDataProvider/07.04.02.SqlDataProvider

```diff
@@ -521,6 +521,174 @@ AS
+	BEGIN
+		CREATE TABLE {databaseOwner}{objectQualifier}OutputCache
+			(
+				[CacheKey] [varchar](36) NOT NULL,
+				[ItemId] [int] NOT NULL,
+				[Data] [ntext] NOT NULL,
+				[Expiration] [datetime] NOT NULL,
+				CONSTRAINT [PK_{objectQualifier}OutputCache] PRIMARY KEY CLUSTERED ( [CacheKey] ASC )
+			)
+	END
+GO
+
+/* Add New OutputCacheGetItem  */
+/*******************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetItem]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItem
+GO
+	
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItem
+	@CacheKey [varchar](36)
+AS
+    SELECT *
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE CacheKey = @CacheKey
+GO
+
+/* Add New OutputCacheAddItem  */
+/*******************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheAddItem]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheAddItem
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheAddItem
+	@ItemId int,
+	@CacheKey [varchar](36),
+	@Data	ntext,
+	@Expiration	datetime
+AS
+    INSERT INTO {databaseOwner}{objectQualifier}OutputCache
+		(ItemId, 
+		CacheKey, 
+		Data, 
+		Expiration)
+    VALUES
+		(@ItemId, 
+		@CacheKey, 
+		@Data, 
+		@Expiration)
+GO
+
+/* Add New OutputCacheGetItemCount  */
+/************************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetItemCount]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+	@ItemId int
+AS
+    SELECT COUNT(*)
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE ItemId = @ItemId
+GO
+
+/* Add New OutputCachePurgeCache  */
+/**********************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCachePurgeCache]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeCache
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeCache
+AS
+    DELETE
+    FROM {databaseOwner}{objectQualifier}OutputCache
+GO
+
+/* Add New OutputCachePurgeExpiredItems  */
+/*****************************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCachePurgeExpiredItems]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeExpiredItems
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeExpiredItems
+	@CurrentUtcDateTime datetime
```

Comment:
currentDateTime in code behind should be taken from the database (GetDatabaseTimeUTC), therefore you don't need to pass it back. Just compare to GetUTCDate()